pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -189,3 +190,11 @@ func (p *InstanceTypeProvider) CacheUnavailable(ctx context.Context, fleetErr *e
 func UnavailableOfferingsCacheKey(capacityType string, instanceType string, zone string) string {
 	return fmt.Sprintf("%s:%s:%s", capacityType, instanceType, zone)
 }
+
+func InstanceTypesCacheKey(provider *v1alpha1.AWS) string {
+	providerHash, err := hashstructure.Hash(provider, hashstructure.FormatV2, nil)
```

Comment:
Totally optional, but I avoid qualifying a variable with multiple words unless it helps clarify. In this case, it's unambiguous given the context. 
```suggestion
	hash, err := hashstructure.Hash(provider, hashstructure.FormatV2, nil)
```